agent/src/main/scala/za/co/absa/atum/agent/model/Measure.scala

```diff
@@ -48,86 +47,86 @@ object AtumMeasure {
 
-    override def function: MeasurementFunction =
+    override def function: MeasurementProcessor.MeasurementFunction =
       (ds: DataFrame) => {
         val resultValue = ds.select(columnExpression).collect()
-        ResultOfMeasurement(resultValue(0).toString, resultValueType)
+        MeasureResult(resultValue(0).toString, resultValueType)
       }
 
-    override def controlColumns: Seq[String] = Seq.empty
+    override def measuredColumns: Seq[String] = Seq.empty
     override val resultValueType: ResultValueType.ResultValueType = ResultValueType.Long
   }
   object RecordCount {
     private[agent] val measureName: String = "count"
     def apply(): RecordCount = RecordCount(measureName)
   }
 
-  case class DistinctRecordCount private (measureName: String, controlCols: Seq[String]) extends AtumMeasure {
-    require(controlCols.nonEmpty, "At least one control column has to be defined.")
+  case class DistinctRecordCount private (measureName: String, measuredCols: Seq[String]) extends AtumMeasure {
+    require(measuredCols.nonEmpty, "At least one measured column has to be defined.")
 
-    private val columnExpression = countDistinct(col(controlCols.head), controlCols.tail.map(col): _*)
+    private val columnExpression = countDistinct(col(measuredCols.head), measuredCols.tail.map(col): _*)
 
-    override def function: MeasurementFunction =
+    override def function: MeasurementProcessor.MeasurementFunction =
       (ds: DataFrame) => {
         val resultValue = ds.select(columnExpression).collect()
-        ResultOfMeasurement(resultValue(0)(0).toString, resultValueType)
+        MeasureResult(resultValue(0)(0).toString, resultValueType)
       }
 
-    override def controlColumns: Seq[String] = controlCols
+    override def measuredColumns: Seq[String] = measuredCols
     override val resultValueType: ResultValueType.ResultValueType = ResultValueType.Long
   }
   object DistinctRecordCount {
     private[agent] val measureName: String = "distinctCount"
-    def apply(controlCols: Seq[String]): DistinctRecordCount = DistinctRecordCount(measureName, controlCols)
+    def apply(measuredCols: Seq[String]): DistinctRecordCount = DistinctRecordCount(measureName, measuredCols)
   }
 
-  case class SumOfValuesOfColumn private (measureName: String, controlCol: String) extends AtumMeasure {
+  case class SumOfValuesOfColumn private (measureName: String, measuredCol: String) extends AtumMeasure {
     private val columnAggFn: Column => Column = column => sum(column)
 
-    override def function: MeasurementFunction = (ds: DataFrame) => {
-      val dataType = ds.select(controlCol).schema.fields(0).dataType
-      val resultValue = ds.select(columnAggFn(castForAggregation(dataType, col(controlCol)))).collect()
-      ResultOfMeasurement(handleAggregationResult(dataType, resultValue(0)(0)), resultValueType)
+    override def function: MeasurementProcessor.MeasurementFunction = (ds: DataFrame) => {
```

Comment:
Why do you prefer `MeasurementProcessor.MeasurementFunction` over `MeasurementFunction`?

Reply:
No good idea and I can't remember, I would say that it's unintentional side effect of refactoring. I'll certainly change it